src/main/groovy/com/ullink/ProjectFileParser.groovy

```diff
@@ -44,21 +44,42 @@ class ProjectFileParser {
         FilenameUtils.removeExtension(file) + newExtension
     }
 
+    File getOutputPath() {
+        if (msbuild.destinationDir) {
+            project.file(msbuild.destinationDir)
+        }
+        if (properties.TargetDir) {
+            new File (properties.TargetDir)
+        }
+    }
+
     File getDotnetAssemblyFile() {
-        project.file(properties.TargetPath)

```

Comment:
But it should not, cf [here](https://github.com/ngyukman/gradle-msbuild-plugin/blob/master/src/main/groovy/com/ullink/Msbuild.groovy#L106) allProjects doesn't contain the solution, only its projects. What's your build file looking like ?